packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -1501,32 +1927,52 @@ fn _run(
     let runtime = tokio::runtime::Builder::new_current_thread()
         .enable_all()
         .build()
-        .map_err(|e| Error::IO("Local tokio runtime".into(), e))?;
+        .map_err(|err| Error::IO("Local tokio runtime".into(), err))?;
 
     tokio::pin! {
         let impl_future = async {
-            let mv8 = MiniV8::new();
-            let mut impl_ = RunnerImpl::new(&mv8, &init_msg)?;
+            let platform = v8::new_default_platform(0, false).make_shared();
+            v8::V8::initialize_platform(platform);
+            v8::V8::initialize();
+
+            let isolate = &mut v8::Isolate::new(Default::default());
+
+            let scope = &mut v8::HandleScope::new(isolate);
+            let context = v8::Context::new(scope);
+            let scope = &mut v8::ContextScope::new(scope, context);
+
+            let mut thread_local_runner = ThreadLocalRunner::new(scope, context, &init_msg)?;
+
             loop {
                 tokio::select! {
                     Some((span, sim_id, msg)) = inbound_receiver.recv() => {
                         let _span = span.entered();
                         // TODO: Send errors instead of immediately stopping?
                         let msg_str = msg.as_str();
-                        tracing::debug!("JS runner got sim `{:?}` inbound {}", &sim_id, msg_str);
-                        let keep_running = impl_.handle_msg(&mv8, sim_id, msg, &outbound_sender)?;
-                        tracing::debug!("JS runner handled sim `{:?}` inbound {}", sim_id, msg_str);
+                        tracing::debug!("JS runner got sim `{sim_id:?}` inbound {msg_str}");
+                        let keep_running = thread_local_runner.handle_msg(scope, sim_id, msg, &outbound_sender)?;
```

Comment:
```suggestion
                        let keep_running = thread_local_runner.handle_msg(
                            &mut context_scope,
                            sim_id,
                            msg,
                            &outbound_sender,
                        )?;
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/228f6ea59b829f94abe4faaa64a78415a6fcca9b